src/scanpy/preprocessing/_simple.py

```diff
@@ -754,7 +754,7 @@ def regress_out(
     if not variable_is_categorical and np.linalg.det(regressors.T @ regressors) != 0:
         if np.issubdtype(X.dtype, np.integer):
             target_dtype = np.float32 if X.dtype.itemsize <= 4 else np.float64
-            X = X.astype(target_dtype)
+            X = X.astype(target_dtype, order="C")
```

Comment:
But this could also apply to sparse? And we don't want the argument then, no? https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.astype.html nothing here about `order`

Reply:
indeed it was a bug, tested for and resolved